assets/javascripts/modules/autoComplete.js

```diff
@@ -314,28 +314,104 @@ var suggestionsEvent = function () {
   });
 };
 
+
+/**
+  * Generate random variable name to ensure no conflicts
+*/
+var randomVariableName = function () {
+  var text = "suggestions";
+  var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  for( var i=0; i < 5; i++ ) {
+      text += possible.charAt(Math.floor(Math.random() * possible.length));
+  }
+  return text;
+};
+
+/**
+  * Either grab a given global variable or generate one from a given select input
+*/
+var getSuggestions = function (el) {
+  var suggestions;
+  var dataLinked = $('select[id="' + el.attr("data-linked") + '"]');
+  if(dataLinked.length > 0){
+    // using a select input in-page as the source for the suggestions
+    // this is useful when using a select as a fallback for the autocomplete
+    suggestions = [];
+    dataLinked.find('option').each(function(){
+      if($(this).text() > ""){
+        suggestions.push({"title":$(this).text(), "value":$(this).attr("value")});
+      }
+    });
+    // need to populate the data-suggestions attribute from this for validation
+    // generate a global variable to hold the data and insert into data-suggestions attribute
+    var r = randomVariableName();
+    window[r] = suggestions;
+    el.attr("data-suggestions", r);
+  } else {
+    // using a global js variable which has already been defined
+    if(window.countries){
+      // legacy fallback for original countries global variable
+      suggestions = countries;
+    } else {
+      // allow custom variable
+      suggestions = window[el.attr("data-suggestions")];
+    }
+  }
+  return suggestions;
+};
+
+/**
+  * Find input to post value back to
+  * See if there is a legacy input, else use supplied one
+*/
+var getTargetInput = function (el) {
+  var target;
+  // input to post back a selection from the autocomplete
+  if($('#countryCode').length > 0){
+    // legacy support for original
+    target = $('#countryCode');
+  } else {
+    var dataLinked = $('[id="' + el.attr("data-linked") + '"]');
+    if(dataLinked.length > 0){
+      target = dataLinked;
+    }
+  }
+  return target;
+};
```

Comment:
I think my comment has been misunderstood. Keep the original instantiation and in the `autoCompleteFactory.js` and inside of `autoComplete.js` check if the argument is an element, if not then check the `data-linked` attribute
>Then you can check to see if it is an element or a string and do the appropriate.